### ISSUE TITLE
Docker machine fails to create a machine on timeout on getting the docker releases

### ISSUE THREAD

**Issue by delgurth** [docker/machine#2186]
I'm trying to create a docker-machine, but it fails to do so when I do not have the boot2docker image stored in the cache (in ~/.docker/machine/cache):

```
docker-machine -D create --driver=virtualbox broken                                                                                                                                                                                   Thu Nov  5 14:50:41 2015
Docker Machine Version: 0.5.0 (04cfa58)
Found binary path at /usr/local/bin/docker-machine-driver-virtualbox
Launching plugin server for driver virtualbox
Plugin server listening at address 127.0.0.1:57490
() Calling RpcServerDriver.GetVersion
Using API Version 1
() Calling RpcServerDriver.SetConfigRaw
() Calling RpcServerDriver.GetMachineName
(flag-lookup) Calling RpcServerDriver.GetCreateFlags
Making call to close connection to plugin binary
Making call to close driver server
(flag-lookup) Calling RpcServerDriver.Close
Successfully made call to close driver server
Found binary path at /usr/local/bin/docker-machine-driver-virtualbox
Launching plugin server for driver virtualbox
Plugin server listening at address 127.0.0.1:57494
() Calling RpcServerDriver.GetVersion
Using API Version 1
() Calling RpcServerDriver.SetConfigRaw
() Calling RpcServerDriver.GetMachineName
(broken) Calling RpcServerDriver.GetMachineName
(broken) Calling RpcServerDriver.DriverName
(broken) Calling RpcServerDriver.GetCreateFlags
(broken) Calling RpcServerDriver.SetConfigFromFlags
Running pre-create checks...
(broken) Calling RpcServerDriver.PreCreateCheck
(broken) DBG | COMMAND: /usr/local/bin/VBoxManage
(broken) Calling RpcServerDriver.GetConfigRaw
(broken) Calling RpcServerDriver.GetConfigRaw
Creating machine...
(broken) Calling RpcServerDriver.Create
(broken) OUT | No default boot2docker iso found locally, downloading the latest release...
Error creating machine: Error in driver during machine creation: Get https://api.github.com/repos/boot2docker/boot2docker/releases: dial tcp: i/o timeout
```

Just getting it using wget worked, but it was slow on resolving the host. Unfortunately you do not see timings in the output. But perhaps this i/o timeout should be increased or in case it keeps failing you send a message to the user on how to download it manually.

```
wget  https://api.github.com/repos/boot2docker/boot2docker/releases                                                                                                                                                               Thu Nov  5 14:50:11 2015
--2015-11-05 14:50:22--  https://api.github.com/repos/boot2docker/boot2docker/releases
Resolving api.github.com... 192.30.252.126
Connecting to api.github.com|192.30.252.126|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 117611 (115K) [application/json]
Saving to: 'releases'

releases                                                            100%[=====================================================================================================================================================================>] 114.85K   376KB/s   in 0.3s

2015-11-05 14:50:26 (376 KB/s) - 'releases' saved [117611/117611]
```

Also, when I tried to download it later it fails because too many people used this api call to get the releases (we are in a workshop by @arun-gupta). So perhaps it's not wise to depend on this API call to determine the latest version:

```
{
  "message": "API rate limit exceeded for 92.70.102.178. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
  "documentation_url": "https://developer.github.com/v3/#rate-limiting"
}
```

**Comment by dgageot** [comment]
Same problem as https://github.com/docker/machine/issues/2167

**Comment by dgageot** [comment]
closed this in 1f0de8f
